test/emqx_shared_sub_SUITE.erl

```diff
@@ -37,11 +37,15 @@ all() -> emqx_ct:all(?SUITE).
 init_per_suite(Config) ->
     net_kernel:start(['master@127.0.0.1', longnames]),
     emqx_ct_helpers:boot_modules(all),
+    PortDiscovery = application:get_env(gen_rpc, port_discovery),
+    application:set_env(gen_rpc, port_discovery, stateless),
+    application:ensure_all_started(gen_rpc),
     emqx_ct_helpers:start_apps([]),
-    Config.
+    [{port_discovery, PortDiscovery} | Config].
 
-end_per_suite(_Config) ->
-    emqx_ct_helpers:stop_apps([]).
+end_per_suite(Config) ->
+    emqx_ct_helpers:stop_apps([gen_rpc]),
+    application:set_env(gen_rpc, port_discovery, proplists:get_value(port_discovery, Config)).
```

Comment:
I think this check still needs to happen, since `application:get_env/2` wraps the value in `{ok, _}`.

```erlang
1> application:get_env(app, key).
undefined
2> application:set_env(app, key, value).
ok
3> application:get_env(app, key).       
{ok,value}
```


```suggestion
    case proplists:get_value(port_discovery, Config) of
      {ok, OldValue} -> application:set_env(gen_rpc, port_discovery, OldValue);
      _ -> ok
    end.
```